addons/sale_project/models/project.py

```diff
@@ -76,14 +76,15 @@ def action_create_invoice(self):
 
     def _get_stat_buttons(self):
         buttons = super(Project, self)._get_stat_buttons()
-        buttons.append({
-            'icon': 'dollar',
-            'text': _('Sales Order'),
-            'action_type': 'object',
-            'action': 'action_view_so',
-            'show': self.user_has_groups('sales_team.group_sale_salesman_all_leads') and bool(self.sale_order_id),
-            'sequence': 1,
-        })
+        if self.user_has_groups('sales_team.group_sale_salesman_all_leads'):
+            buttons.append({
+                'icon': 'dollar',
+                'text': _('Sales Order'),
+                'action_type': 'object',
+                'action': 'action_view_so',
+                'show': self.user_has_groups('sales_team.group_sale_salesman_all_leads') and bool(self.sale_order_id),
```

Comment:
Since we are sure the user has the group, you can directly do this:
```suggestion
                'show': bool(self.sale_order_id),
```